tools/tokamak_grids/gridgen/process_grid.pro

```diff
@@ -1619,6 +1619,45 @@ retrybetacalc:
   psi_bndry = mesh.faxis + mesh.fnorm
   s = file_write(handle, "psi_bndry", psi_bndry)
 
+  ; save some version information
+  ;
+  ; BOUT++ version information: this is set when BOUT++ is configured.
+  ; Hypnotoad doesn't require BOUT++ to have been configured, and IDL code may
+  ; have changed since 'configure' was run, so this is not 100% reliable, but
+  ; still worth saving as a sanity check
+  hypnotoad_info = ROUTINE_INFO('hypnotoad', /SOURCE)
+  hypnotoad_path = FILE_DIRNAME(hypnotoad_info.path)
+
+  ; BOUT++ git hash
+  ;; old version got git hash from bout-config, but this may be out of date
+  ;SPAWN, STRJOIN([hypnotoad_path, PATH_SEP(), '..', PATH_SEP(), '..', PATH_SEP(), '..', PATH_SEP(), 'bin/bout-config --git']), bout_git_hash, EXIT_STATUS=status
+  ;IF status THEN BEGIN
+  ;  PRINT, "WARNING: Failed to get Git hash for BOUT++, could not run '../../../bin/bout-config --git'. Have you configured BOUT++ successfully?"
+  ; new version gets current git hash directly from git-describe
+  SPAWN, STRJOIN(['cd ',hypnotoad_path, '&& git describe --always --abbrev=0 --dirty --match "NOT A TAG"']), bout_git_hash, EXIT_STATUS=status
+  IF status THEN BEGIN
+    PRINT, 'WARNING: Failed to get Git hash for BOUT++, could not run < git describe --always --abbrev=0 --dirty --match "NOT A TAG" > in Hypnotoads directory.'
+  ENDIF ELSE BEGIN
+    ; bout_git_hash as returned from SPAWN seems to have some funny character
+    ; in, maybe a trailing newline. This character causes an error when trying
+    ; to write as a NetCDF attribute. STRJOIN seems to fix this.
+    bout_git_hash = STRJOIN(bout_git_hash, '')
+
+    s = file_write_attribute(handle, "git_hash", bout_git_hash)
+  ENDELSE
+
+  ; BOUT++ version number
+  SPAWN, STRJOIN([hypnotoad_path, PATH_SEP(), '..', PATH_SEP(), '..', PATH_SEP(), '..', PATH_SEP(), 'bin/bout-config --version']), bout_version, EXIT_STATUS=status
+  IF status THEN BEGIN
+    PRINT, "WARNING: Failed to get version number for BOUT++, could not run '../../../bin/bout-config --version'. Have you configured BOUT++ successfully?"
```

Comment:
Would it be better to try and call `bout-config` without a relative path, and ask the user to check it's in `$PATH`?

Reply:
The warning needs improving... The path it actually uses is relative to the location of the Hypnotoad that is actually running, which seems sensible to me, unless BOUT++ is installed. If its installed I guess we can assume that bout-config should be in the `$PATH`. I'll try to update to handle:
* normal git repo: get git hash from `git describe`, version from `bout-config` using relative path
* tarball download: get git hash and version from `bout-config` using relative path
* installed as library: get both from `bout-config` in `$PATH`, giving an informative warning if they can't be found.